### PR TITLE
Moved resize image to config

### PR DESCRIPTION
**Pull request by zhenghh04**
This is to reduce the overhead for reader inititialization.

### CODE REVIEW

File: src/reader/reader_handler.py [hunk 1]
```diff
@@ -29,26 +29,32 @@
...
         logging.debug(
             f"Loading {self.__class__.__qualname__} reader on thread {self.thread_index} from rank {self._args.my_rank}")
         self.dataset_type = dataset_type
         self.open_file_map = {}
-        self.epoch_number = epoch_number
+
         if FormatReader.read_images is None:
             FormatReader.read_images = 0
-
+        self.image = None
```
Comment (hariharan-devarajan) [inline src/reader/reader_handler.py:50]: Now that we dont need to share image between get_sample and resize. Can u remove this and make it a local variable in the get_sample method?

File: src/reader/tf_reader.py [hunk 2]
```diff
@@ -73,38 +66,44 @@ def _tf_parse_function(self, serialized):
...
+        parsed_example = tf.io.parse_example(serialized=serialized, features=features)
+        # Get the image as raw bytes.
+        image_raw = parsed_example['image']
+        dimension = tf.cast(parsed_example['size'], tf.int32).numpy()
+        # Decode the raw bytes so it becomes a tensor with type.
+        image_tensor = tf.io.decode_raw(image_raw, tf.float64)
+        size = dimension * dimension
+        dlp.update(image_size=size)
+        # image_tensor = tf.io.decode_image(image_raw)
+        resized_image = tf.reshape(image_tensor, [dimension, dimension])
```
Comment (hariharan-devarajan) [inline src/reader/tf_reader.py:78]: We have to have a similar logic here not to reshape but simulate preprocess and return resized image.

File: src/reader/tf_reader.py [hunk 3]
```diff
@@ -73,38 +66,44 @@ def _tf_parse_function(self, serialized):
...
-        val = super().read_index(index)
-        t1 = time()
-        PerfTrace.get_instance().event_complete(
-            f"{self.read_index.__qualname__}", MY_MODULE, t0, t1 - t0, arguments=self.profile_args)
-        return val
-
-    @event_logging(module=MY_MODULE, arguments=profile_args)
+        total = math.ceil(len(_file_list)/self._args.comm_size / batch_size)
+        step = 1
+        for batch in dlp.iter(self._dataset):
```
Comment (hariharan-devarajan) [inline src/reader/tf_reader.py:96]: Do u want to keep the dlp.iter only here?

**Review by hariharan-devarajan — CHANGES_REQUESTED** [review]
@zhenghh04 I think some minor fixes are needed and then u can merge.

File: src/reader/tf_reader.py (hunk 3 above)
Comment (zhenghh04) [reply src/reader/tf_reader.py:96]: I updated the change and removed dlp.iter from Line 96 also.

File: src/reader/tf_reader.py (hunk 2 above)
Comment (zhenghh04) [reply src/reader/tf_reader.py:78]: I returned a tf.convert_to_tensor(resized_image)

**Review by hariharan-devarajan — APPROVED** [review]

File: src/data_generator/csv_generator.py [hunk 4]
```diff
@@ -64,8 +69,7 @@ def generate(self):
             df.to_csv(out_path_spec, compression=compression)
         MPI.COMM_WORLD.Barrier()
         source_out_path_spec = self.storage.get_uri(self._file_list[0])
-        for i in range(1, int(self.total_files_to_generate)):
-            if i % self.comm_size == self.my_rank:
-                progress(i+1, self.total_files_to_generate, "Generating CSV Data")
-                out_path_spec = self.storage.get_uri(self._file_list[i])
-                copyfile(source_out_path_spec, out_path_spec)
+        for i in dlp.iter(range(self.my_rank, int(self.total_files_to_generate), self.comm_size)):
```
Comment (hariharan-devarajan) [inline src/data_generator/csv_generator.py:72]: The test is failing due to this. It's copying the same file. u have to skip that file for rank 0 as its already created.

**Review by hariharan-devarajan — CHANGES_REQUESTED** [review]

**Review by hariharan-devarajan — APPROVED** [review]

**Comment by hariharan-devarajan** [comment]
You can merge the PR.

**Comment by hariharan-devarajan** [comment]
Also includes changes from #54

**Pull request merged**